src/components/cart/DeliveryCheck.tsx

```diff
@@ -1,24 +1,33 @@
 import { useState } from 'react';
+import { useRecoilState } from 'recoil';
 import { styled } from 'styled-components';
 
 import { CheckBigCheckedIcon, CheckBigNotCheckedIcon, ShoppingCartIcon } from '../../assets/icon';
+import { overallCheckState } from '../../states/cart';
 
 const DeliveryCheck = () => {
   const [deliveryCheck, setDeliveryCheck] = useState(true);
   const [progressAmount, setProgressAmount] = useState('100%');
+
+  const [overallCheck, setOverallCheck] = useRecoilState(overallCheckState);
+
+  const onInputChange = () => {
+    Object.keys(overallCheck).forEach((key) => {
+      setOverallCheck((prevOverallCheck) => ({
+        ...prevOverallCheck,
+        [key]: !deliveryCheck,
+      }));
+    });
```

Comment:
```suggestion
    Object.keys(overallCheck).forEach((key) => {
    if(overallCheck[key] !== deliveryCheck) {
      setOverallCheck((prevOverallCheck) => ({
        ...prevOverallCheck,
        [key]: !deliveryCheck,
      }));
      }
    });
```

여기서는 큰 성능 차이를 바꾸지는 않겠지만 이렇게 바꿔줘야할 애들만 골라서 넣어주면 좀 더 효율적이겠군요!!